diff --git a/Makefile.libretro b/Makefile.libretro
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -423,6 +423,33 @@ else ifeq ($(platform), miyoo)
    CFLAGS += -fno-common -ftree-vectorize -funswitch-loops
    DEFINES += -std=c99
 
+# MIYOOMINI
+else ifeq ($(platform), miyoomini)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   SHARED  := -shared -Wl,--version-script=link.T -Wl,--no-undefined 
+   fpic    := -fPIC
+   PLATFORM_DEFINES += -std=c99 -D_GNU_SOURCE -DHAVE_STRTOF_L -DHAVE_LOCALE -DHAVE_NO_LANGEXTRA
+   PLATFORM_DEFINES += -marm -march=armv7ve+simd -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
+   DEFINES  += -Ofast -fPIC -flto=4 -fipa-pta -fipa-ra -fuse-linker-plugin
+   DEFINES  += -falign-functions=1 -falign-jumps=1 -falign-loops=1
+   DEFINES  += -fno-stack-protector -fno-ident -fomit-frame-pointer
+   DEFINES  += -fno-unwind-tables -fno-asynchronous-unwind-tables
+   DEFINES  += -fmerge-all-constants -fno-math-errno -ffast-math 
+   DEFINES  += -ftree-vectorize -funswitch-loops -funroll-loops -fno-common
+   DEFINES  += -fdata-sections -ffunction-sections -Wl,-s -Wl,--gc-sections
+   LDFLAGS  += -fPIC -flto=4 -fipa-pta -fipa-ra
+   LDFLAGS  += -Wl,-s -Wl,--gc-sections   
+   ARCH = arm
+   HAVE_NEON = 1
+   ifeq ($(TARGET_NAME), mgba_sgb)
+      M_CORE_GB  = 1
+      M_CORE_GBA = 0
+   endif
+
+
 # Windows
 else
    TARGET := $(TARGET_NAME)_libretro.dll
